App/Services/UserNotifier.cs

```diff
@@ -20,17 +21,40 @@ public interface IUserNotifier : INotificationHandler, IAsyncDisposable
     public void UnregisterHandler(string name);
 
     public Task ShowErrorNotification(string title, string message, CancellationToken ct = default);
-    public Task ShowActionNotification(string title, string message, string handlerName, IDictionary<string, string>? args = null, CancellationToken ct = default);
+    /// <summary>
+    /// This method allows to display a Windows-native notification with an action defined in
+    /// <paramref name="handlerName"/> and provided <paramref name="args"/>.
+    /// </summary>
+    /// <param name="title">Title of the notification.</param>
+    /// <param name="message">Message to be displayed in the notification body.</param>
+    /// <param name="handlerName">Handler should be e.g. <c>nameof(Handler)</c> where <c>Handler</c>
+    /// implements <see cref="Coder.Desktop.App.Services.INotificationHandler" />.
+    /// If handler is <c>null</c> the action will open Coder Desktop.</param>
+    /// <param name="args">Arguments to be provided to the handler when executing the action.</param>
+    public Task ShowActionNotification(string title, string message, string? handlerName, IDictionary<string, string>? args = null, CancellationToken ct = default);
 }
 
-public class UserNotifier(ILogger<UserNotifier> logger, IDispatcherQueueManager dispatcherQueueManager) : IUserNotifier
+public class UserNotifier : IUserNotifier
 {
     private const string CoderNotificationHandler = "CoderNotificationHandler";
+    private const string DefaultNotificationHandler = "DefaultNotificationHandler";
 
     private readonly AppNotificationManager _notificationManager = AppNotificationManager.Default;
+    private readonly ILogger<UserNotifier> _logger;
+    private readonly IDispatcherQueueManager _dispatcherQueueManager;
 
     private ConcurrentDictionary<string, INotificationHandler> Handlers { get; } = new();
 
+    public UserNotifier(ILogger<UserNotifier> logger, IDispatcherQueueManager dispatcherQueueManager,
+        INotificationHandler notificationHandler)
```

Comment:
Perhaps we should wrap it with a marker type like `public interface IDefaultNotificationHandler : INotificationHandler` so we don't inadvertently add another implementation that could get DI'd here, and only have the `App` implement `IDefaultNotificationHandler`